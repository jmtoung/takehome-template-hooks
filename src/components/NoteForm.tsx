import React from 'react';
import { Form, Button } from 'react-bootstrap';

export const NoteForm: React.FunctionComponent<{
  title: string;
  body: string;
  text: string;
  handleBodyChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  handleTitleChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  handleSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
}> = (props) => {
  const {
    title,
    body,
    text,
    handleSubmit,
    handleBodyChange,
    handleTitleChange,
  } = props;

  return (
    <Form onSubmit={handleSubmit}>
      <Form.Group controlId="exampleForm.ControlTextarea1">
        <Form.Control
          type="text"
          placeholder="Title..."
          className="title-input"
          name="title"
          value={title}
          onChange={handleTitleChange}
        />
        <Form.Control
          as="textarea"
          rows={4}
          placeholder="Content..."
          name="body"
          className="body-input"
          value={body}
          onChange={handleBodyChange}
        />
      </Form.Group>
      <Button variant="primary" type="submit">
        {text}
      </Button>
    </Form>
  );
};
