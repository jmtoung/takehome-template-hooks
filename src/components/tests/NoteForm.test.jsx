import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import NoteForm from '../NoteForm';

describe('NoteForm', () => {
  it('renders noteForm component correctly.', async () => {
    const wrapper = shallow(<NoteForm
      title="Testing Title"
      body="Testing Body"
      text="Testing"
      handleChange={jest.fn}
      handleSubmit={jest.fn}
    />);

    expect(toJSON(wrapper)).toMatchSnapshot();

  });

  it('calls handleSubmit when form is submitted.', () => {
    const onSubmitFn = jest.fn();
    const wrapper = shallow(<NoteForm handleSubmit={onSubmitFn} />);
    const form = wrapper.find('Form');
    form.simulate('submit');
    expect(onSubmitFn).toHaveBeenCalledTimes(1);
  });
  
  it('calls handleChange when title input is changed.', () => {
    const onChangeFn = jest.fn();
    const wrapper = shallow(<NoteForm handleChange={onChangeFn} />);
    const input = wrapper.find('.title-input');
    input.simulate('change');
    expect(onChangeFn).toHaveBeenCalledTimes(1);
  });
  
  it('calls handleChange when body input is changed.', () => {
    const onChangeFn = jest.fn();
    const wrapper = shallow(<NoteForm handleChange={onChangeFn} />);
    const input = wrapper.find('.body-input');
    input.simulate('change');
    expect(onChangeFn).toHaveBeenCalledTimes(1);
  });
});