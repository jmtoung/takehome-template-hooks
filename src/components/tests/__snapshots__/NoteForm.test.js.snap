// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteForm renders noteForm component correctly. 1`] = `
<Form
  inline={false}
  onSubmit={[Function]}
>
  <FormGroup
    controlId="exampleForm.ControlTextarea1"
  >
    <FormControl
      className="title-input"
      name="title"
      placeholder="Title..."
      type="text"
      value="Testing Title"
    />
    <FormControl
      as="textarea"
      className="body-input"
      name="body"
      placeholder="Content..."
      rows={4}
      value="Testing Body"
    />
  </FormGroup>
  <Button
    active={false}
    disabled={false}
    type="submit"
    variant="primary"
  >
    Testing
  </Button>
</Form>
`;
